### PR TITLE
Session migration

### PR DESCRIPTION
**Pull request by cnderrauber**

### CODE REVIEW

File: livekit/types.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ package livekit
 type TrackID string
 
 func StringsAsTrackIDs(trackIDs []string) []TrackID {
-	var asTrackID []TrackID
+	asTrackID := make([]TrackID, 0, len(trackIDs))
```
Comment (boks1971) [inline livekit/types.go:8]: Probably can make the same change in room names function at the bottom on this file to be consistent.

File: livekit_rtc.proto [hunk 2]
```diff
@@ -217,3 +219,9 @@ message SubscriptionPermissionUpdate {
   string track_sid = 2;
   bool allowed = 3;
 }
+
+message SyncState {
+  SessionDescription anwser = 1;
```
Comment (boks1971) [inline livekit_rtc.proto:224]: Can we name this `subscriber_answer` to indicate that it corresponds to subscriber peer connection?

**Review by boks1971 — APPROVED** [review]
A couple of small comments

File: livekit_internal.proto [hunk 3]
```diff
@@ -107,6 +107,7 @@ message StartSession {
   bool recorder = 12;
   ClientInfo client = 11;
   string name = 13;
+  bool migrate = 14;
```
Comment (davidzhao) [inline livekit_internal.proto:110]: is this needed? or could we determine from protocol_version + which node is receiving the message?

Comment (cnderrauber) [reply livekit_internal.proto:110]: removed

File: livekit_rtc.proto (hunk 2 above)
Comment (cnderrauber) [reply livekit_rtc.proto:224]: In our SignalRequest, answer is only used for subscribe peer connection, so use answer for consistent

File: livekit/types.go (hunk 1 above)
Comment (cnderrauber) [reply livekit/types.go:8]: Done

**Pull request merged**